### PR TITLE
[v47] Mint Params Migrate

### PR DESCRIPTION
**Pull request by Reecepbcups**
Builds off of https://github.com/CosmosContracts/juno/pull/706

Review starts at https://github.com/CosmosContracts/juno/commit/df7950c3f4e85b609d1020bdfdb9d965a9223c2e

## Files of importance:
- proto/juno/mint/tx.proto
- x/mint/exported/exported.go
- x/mint/keeper/keeper.go
- x/mint/keeper/migrator.go
- x/mint/keeper/msg_server.go
- x/mint/migrations/v3/migrate.go (note: this is v3 and uses paramkey of []byte{0x01})
- x/mint/module.go
- x/mint/types/msgs.go
- x/mint/types/params.go -> x/mint/types/params_legacy.go

### CODE REVIEW

File: app/upgrades/v15/upgrades.go
Comment (dimiandre) [inline app/upgrades/v15/upgrades.go:31]: should be in TF pr?

**Review by dimiandre — APPROVED** [review]
Expect that there are some changes about tokenfactory it looks good

File: app/upgrades/v15/upgrades.go
Comment (Reecepbcups) [reply app/upgrades/v15/upgrades.go:31]: I merged Tf pr into this since they build off each other. TF PR should be reviewed first then main merged to here

**Comment by faddat** [comment]
@dimiandre before I approve, what are the tf changes here that concern you?  Can you let me know if the conflict resolution I do here resolves the issues?

**Comment by Reecepbcups** [comment]
@faddat If you actually read the comments Dimi & I left: https://github.com/CosmosContracts/juno/pull/707#discussion_r1234442100

because this PR was built off the TF PR (per the first line in this thread) changes were merged here to pass test. 

These changes are no longer in here because they are in main.

**Comment by faddat** [comment]
I read them, I wanted to ask, to be cautious. I'm so sorry that you don't feel that I did.

**Review by faddat — APPROVED** [review]
Approved after having my ignorance confirmed

**Pull request merged**
